src/sentry/api/base.py

```diff
@@ -221,9 +222,15 @@ def paginate(
 
 
 class EnvironmentMixin(object):
+    def _get_environment_id_func(self, request, organization_id):
+        return functools.partial(
+            self._get_environment_id_from_request,
+            request,
+            organization_id,
+        )
```

Comment:
@bretthoerner made me do it, don't @ me

Reply:
Haha, the one change I think we should make here is to add a comment explaining the need for 3 states, and why exception throwing was better than silently missing a sentinel. If we don't, I forsee future yelling (maybe at ourselves).

Reply:
This is a good point, will add